src/Utils.cpp

```diff
@@ -31,15 +31,33 @@ std::optional<std::string> getAncestorPath(const std::string& path, const std::s
     // Append a "/" to the end of the parentPath to make searching easier
     auto parentPathWithSlash = *parentPath + "/";
 
+    // If the ancestor is the project root we need to rename it, applies only to non-DataModel projects
+    auto normalizedAncestorName = ancestorName;
+    if (rootSourceNode && !isDataModel(parentPathWithSlash) && ancestorName == rootSourceNode->name)
+    {
+        normalizedAncestorName = "ProjectRoot";
+
+        // Set normalizedAncestorName back to ancestorName if it's found in the parentPath before root
```

Comment:
What I meant before is that I think it is simpler to just move this check to after the initial ancestor rfind.
So it is along the lines of

```cpp
// as original
auto ancestor = parentPathWithSlash.rfind(ancestorName+ "/");
if (ancestor != std::string::npos)
{
        ...
}

// at this point we know there is definitely no ancestor with the same name within the path
// we can do a simple check and return ProjectRoot if we see that rootSourceNode.name == ancestorName

if (rootSourceNode && !isDataModel(parentPathWithSlash) && ancestorName == rootSourceNode->name)
{
    return "ProjectRoot"
}
```

Reply:
Oh yeah that makes way more sense